modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/publisher/APIM574ChangeTheStatusOfAnAPIToPrototypedThroughThePublisherRestAPITestCase.java

```diff
@@ -109,132 +100,84 @@ public void testCreateAnAPIThroughThePublisherRest() throws Exception {
-                apiNameTest + "  status not updated as Prototyped");
-
-        //Check whether deprecated API is available in publisher
-        HttpResponse prototypedApiResponse = (apiPublisher.getAPI
-                (apiNameTest, apiProvider, apiVersion));
-        JSONObject prototypedApiObject = new JSONObject(prototypedApiResponse.getData());
-        JSONObject allApiObject = new JSONObject(prototypedApiObject.getString("api"));
-        assertFalse(prototypedApiObject.getBoolean("error"), apiNameTest +
+        restAPIPublisher.changeAPILifeCycleStatus(apiId, APILifeCycleAction.DEPRECATE.getAction());
+        assertTrue(restAPIPublisher.getLifecycleStatus(apiId).getData().equals("Deprecated"), apiNameTest +
+                "  status not updated as Deprecate");
+        //Check whether published API is available in publisher
+        HttpResponse deprecatedApiResponse = restAPIPublisher.getAPI(apiId);
+        assertEquals(deprecatedApiResponse.getResponseCode(), Response.Status.OK.getStatusCode(), apiNameTest +
                 " is not visible in publisher");
-        assertTrue(allApiObject.getString("name").equals(apiNameTest),
-                apiNameTest + " is not visible in publisher");
-        assertTrue(allApiObject.getString("status").equals("DEPRECATED"),
-                "Status of the " + apiNameTest + "is not a valid status");
-
+        assertTrue(deprecatedApiResponse.getData().contains(apiNameTest), apiNameTest + " is not visible in " +
+                "publisher");
     }
 
     @Test(groups = {"wso2.am"}, description = "Change the status of the API to RETIRED through" +
             " the publisher rest API ", dependsOnMethods = "testChangeTheStatusOfTheAPIToDeprecated")
     public void testChangeTheStatusOfTheAPIToRetired() throws Exception {
 
         //Change the status DEPRECATED to RETIRED
-        APILifeCycleStateRequest updateRequest = new APILifeCycleStateRequest
-                (apiNameTest, apiProvider, APILifeCycleState.RETIRED);
-        updateRequest.setRequireResubscription("true");
-        HttpResponse creationResponse = apiPublisher.changeAPILifeCycleStatus(updateRequest);
-        assertTrue(creationResponse.getData().contains("RETIRED"),
-                apiNameTest + "  status not updated as Prototyped");
-
-        //Check whether deprecated API is available in publisher
-        HttpResponse prototypedApiResponse = (apiPublisher.getAPI
-                (apiNameTest, apiProvider, apiVersion));
-        JSONObject prototypedApiObject = new JSONObject(prototypedApiResponse.getData());
-        JSONObject allApiObject = new JSONObject(prototypedApiObject.getString("api"));
-        assertFalse(prototypedApiObject.getBoolean("error"), apiNameTest +
+        restAPIPublisher.changeAPILifeCycleStatus(apiId, APILifeCycleAction.RETIRE.getAction());
+        assertTrue(restAPIPublisher.getLifecycleStatus(apiId).getData().equals("Retired"), apiNameTest +
```

Comment:
Lets switch the equals and use the constant for Retired

Reply:
fixed